### PR TITLE
[TorchTidy Fix] Don't try to collect strides for non-strided tensors

### PR DESCRIPTION
**Pull request by Gamrix**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #83935

### CODE REVIEW

**Review by robieta — APPROVED** [review]
LGTM. (modulo lints) Thanks for the quick fix.

**Review by robieta — APPROVED** [review]

File: test/test_profiler.py [hunk 1]
```diff
@@ -1285,6 +1285,62 @@ def test_tensor_properties(self):
...
+            torch._C._profiler._ExtraFields_TorchOp)
+
+        self.assertEqual(node.extra_fields.inputs.shapes, [[2, 3], [2, 3], []])
+        self.assertEqual(node.extra_fields.inputs.strides, [[], [], []])
+
+        input_info = node.extra_fields.inputs
+
+        # FIXME: Different systems have different names for int64_t
+        # self.assertEqual(input_info.dtypes, ['long long', 'long long', 'Scalar'])
+        self.assertIn(input_info.dtypes[0], ["long long", "long int", "long"])
```
Comment (Gamrix) [inline test/test_profiler.py:1311]: @robieta It likely makes sense change the linked line below to be "Tensor" instead given that the metadata already has the dtype of the tensor, and that string this gives for an int64_t tensor is system dependent.

However, that is outside the scope of this diff.
https://github.com/pytorch/pytorch/blob/09e837634bc76c4380db7119e0e997816d584844/torch/csrc/profiler/collection.cpp#L111

**Review by slgong-fb — APPROVED** [review]
Briefly, the fix is checking tensors have stride and applied the associated codes to them?

**Comment by Gamrix** [comment]
> Briefly, the fix is checking tensors have stride and applied the associated codes to them?

Specifically only recording strides for tensors that have a strided layout (eg sparse tensors have no strides)
